### PR TITLE
RUMM-788 Add data scrubbing API

### PR DESCRIPTION
**Pull request by ncreated**
### What and why?

🚚 This PR adds RUM events scrubbing feature with set of 4 new public APIs:
```swift
public func setRUMViewEventMapper(_ mapper: @escaping (RUMViewEvent) -> RUMViewEvent?)
public func setRUMResourceEventMapper(_ mapper: @escaping (RUMResourceEvent) -> RUMResourceEvent?)
public func setRUMActionEventMapper(_ mapper: @escaping (RUMActionEvent) -> RUMActionEvent?)
public func setRUMErrorEventMapper(_ mapper: @escaping (RUMErrorEvent) -> RUMErrorEvent?)
```

Each API allows the user to inspect and modify RUM events before they get send. Returning `nil` drops the event.

### How?

A basic interface of `EventMapper` is introduced:
```swift
internal protocol EventMapper {
    func map<T: Encodable>(event: T) -> T?
}
```
and plugged into `DataProcessor's` pipeline, just before writing the event to the file.

It is implemented by `RUMEventsMapper` which intercepts each value using closures configured by the user in the public interface.

In the next PR I will also add test scenario to `Example` application with the integration test for scrubbing API.

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference

### CODE REVIEW

File: Sources/Datadog/RUM/Scrubbing/RUMEventsMapper.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    private func map<DM: RUMDataModel>(rumEvent: RUMEvent<DM>, using mapper: (DM) -> DM?) -> RUMEvent<DM>? {
+        if let mappedModel = mapper(rumEvent.model) {
+            var mutableRUMEvent = rumEvent
+            mutableRUMEvent.model = mappedModel
+            return mutableRUMEvent
+        } else {
+            return nil
+        }
+    }
```
Comment (buranmert) [inline Sources/Datadog/RUM/Scrubbing/RUMEventsMapper.swift:59]:
```suggestion
  // let viewEventMapper: RUMViewEventMapper?, ...
    private func map<DM: RUMDataModel>(rumEvent: RUMEvent<DM>, using mapper: ((DM) -> DM?)?) -> RUMEvent<DM>? {
    guard let someMapper = mapper else { return rumEvent }

      if let mappedModel = mapper(rumEvent.model) {
          var mutableRUMEvent = rumEvent
          mutableRUMEvent.model = mappedModel
          return mutableRUMEvent
      } else {
          return nil
      }
  }
```
that way we won't need `noOpMapper` and unnecessary copying of `RUMEvent`s ✅ 
this is already how we do it in `DataProcessor.swift`, we can follow the same pattern in both places

Comment (ncreated) [reply Sources/Datadog/RUM/Scrubbing/RUMEventsMapper.swift]: Indeed, I didn't notice we're copying the event value unnecessarily. Done 👍.

File: Tests/DatadogTests/Datadog/Mocks/RUMDataModelMocks.swift [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+            ),
+            usr: .mockRandom(),
+            view: .init(
+                id: .mockRandom(),
+                referrer: .mockRandom(),
+                url: .mockRandom()
+            )
+        )
+    }
+}
```
Comment (buranmert) [inline Tests/DatadogTests/Datadog/Mocks/RUMDataModelMocks.swift:199]: can we use model generator to create these mocks too?

**Review by buranmert — APPROVED** [review]

File: Tests/DatadogTests/Datadog/Mocks/RUMDataModelMocks.swift (hunk 2 above)
Comment (ncreated) [reply Tests/DatadogTests/Datadog/Mocks/RUMDataModelMocks.swift:199]: I was thinking of this, but it took me `~10 min` to do it by hand, while adding this in `rum-models-generator` would be at least few hours. Let's see how difficult would it be to maintain this hand-made mocks, if too expensive, we can invest our time in `rum-models-generator` 👌.

**Pull request merged**